### PR TITLE
Remove ActionRunner. Refactor PullRequestActor

### PR DESCRIPTION
**Pull request by andriipatsula**

### CODE REVIEW

File: src/Maestro/Maestro.ContainerApp/Actors/PullRequestActor.cs [hunk 1]
```diff
@@ -12,102 +12,36 @@
...
-        }
-
-        public Task UpdateAssetsAsync(Guid subscriptionId, int buildId, string sourceRepo, string sourceSha, List<Asset> assets)
-        {
-            return Implementation.UpdateAssetsAsync(subscriptionId, buildId, sourceRepo, sourceSha, assets);
-        }
-    }
-
-    public abstract class PullRequestActorImplementation : IPullRequestActor, IActionTracker
+    public abstract class PullRequestActor : IPullRequestActor, IActionTracker
```
Comment (premun) [inline src/Maestro/Maestro.ContainerApp/Actors/PullRequestActor.cs]: @andriipatsula should we remove the `IActionTracker` too?

File: src/Maestro/Maestro.ContainerApp/Actors/PullRequestActor.cs [hunk 2]
```diff
@@ -155,14 +89,14 @@ public async Task TrackFailedAction(string action, string result, string method,
...
 
-        public Task<string> RunActionAsync(string method, string arguments)
+        public Task UpdateAssetsAsync(Guid subscriptionId, int buildId, string sourceRepo, string sourceSha, List<Asset> assets)
         {
-            return ActionRunner.RunAction(this, method, arguments);
+            return RunUpdateAssetsAsync(subscriptionId, buildId, sourceRepo, sourceSha, assets);
         }
 
-        Task IPullRequestActor.UpdateAssetsAsync(Guid subscriptionId, int buildId, string sourceRepo, string sourceSha, List<Asset> assets)
+        public Task ProcessPendingUpdatesAsync()
```
Comment (premun) [inline src/Maestro/Maestro.ContainerApp/Actors/PullRequestActor.cs]: I'd just keep one of these

File: src/Maestro/Maestro.ContainerApp/Actors/PullRequestActor.cs (hunk 1 above)
Comment (andriipatsula) [reply src/Maestro/Maestro.ContainerApp/Actors/PullRequestActor.cs]: I don't know. We may want to keep it. If I am not mistaken, it's used for audit..

File: src/Maestro/Maestro.ContainerApp/Actors/PullRequestActor.cs (hunk 2 above)
Comment (andriipatsula) [reply src/Maestro/Maestro.ContainerApp/Actors/PullRequestActor.cs]: Updated

File: src/Maestro/Maestro.ContainerApp/Actors/PullRequestActor.cs (hunk 1 above)
Comment (andriipatsula) [reply src/Maestro/Maestro.ContainerApp/Actors/PullRequestActor.cs]: removed

**Review by premun — APPROVED** [review]

**Pull request merged**